report/arcticreport.Rmd

```diff
@@ -44,103 +44,68 @@ objs <- query_objects(cache_tolerance = 1000000)
-)
+Once it is finished, it will save the output to a txt file saved at `~/report_sizes_info.txt`
+
+
+```{r}
+# Reads in txt file and splits each line by separator 
+size_lines <- scan("~/report_sizes_info.txt", what = "char", sep = "\n")
+split_size_lines <- strsplit(size_lines, " ./")
+
+# Creates empty data frame with 2 columns (size_b: size in bytes, id: id of data folder) removing the total size line at the end
+size_info <- data.frame(size_b = rep(NA, length(split_size_lines)-1),
+                        id = rep(NA, length(split_size_lines)-1))
+
+# Fills the data frame with size and id information
+for (i in 1:(length(split_size_lines)-1)) {
+  size_info$size_b[i] <- split_size_lines[[i]][1]
+  size_info$id[i] <- split_size_lines[[i]][2]
+}
 ```
 
-For the next chunk, just run `stat -c "%.10y %n" * | tr " " ","` This one goes fast, it just gives the timestamps for each folder.
+For the next chunk, just run `stat -c "%.10y %n" * | tr " " "," > ~/report_dates_info.txt` 
+This one goes fast; it just gives the timestamps for each folder, and saves it to a file at `~/report_dates_info.txt`
 
 ```{r}
-date_info <- tribble(~date, ~id,
-"2023-08-17","A2025R",
-"2023-06-15","A2251FM67",
-"2023-06-15","A23X83N25",
-"2023-08-22","A24F1MK7Q",
-"2022-01-26","A24Q7QR39",
-"2022-05-27","A2542J95X",
-"2022-06-22","A25Q4RN5H",
-"2022-11-08","A26688K9D",
-"2022-07-11","A26T0GX63",
-"2021-10-13","A28911S0V",
-"2021-02-11","A2CV4BS5K",
-"2022-01-18","A2G73751P",
-"2022-02-03","A2G737525",
-"2022-01-11","A2H70820F",
-"2022-10-12","A2J38KJ9R",
-"2022-03-14","A2JQ0SW5Z",
-"2023-08-29","A2K06X26P",
-"2022-07-28","A2KW57K4R",
-"2023-07-07","A2KW57K57",
-"2023-08-23","A2MG7FX35",
-"2023-01-04","A2N29P78F",
-"2022-10-11","A2NC5SD71",
-"2021-09-24","A2QF8JK7T",
-"2022-02-03","A2QJ78024",
-"2022-10-19","A2R49GB1J",
-"2021-11-30","A2RV0D21Z",
-"2022-12-21","A2RV0D23X",
-"2022-10-11","A2S46H728",
-"2023-06-15","A2SN0158C",
-"2022-05-26","A2SX64B5F",
-"2023-08-09","A2TB0XX0B",
-"2022-03-15","A2W08WH8M",
-"2021-09-03","A2ZG6G81T",
-"2022-12-08","A2ZK55N82",
-)
+# Reads in txt file and splits each line by separator
+date_lines <- scan("~/report_dates_info.txt", what = "char", sep = "\n")
+split_date_lines <- strsplit(date_lines, ",")
+
+# Creates empty data frame with 2 columns (date: timestamp for each folder, id: id of data folder)
+date_info <- data.frame(date = rep(NA, length(split_date_lines)),
+                        id = rep(NA, length(split_date_lines)))
+
+# Fills the data frame with size and id information
+for (i in 1:length(split_date_lines)) {
+  date_info$date[i] <- split_date_lines[[i]][1]
+  date_info$id[i] <- split_date_lines[[i]][2]
+}
+```
 
 
+```{r}
+# Left joins the size_info and date_info tables
 datasets_add <- left_join(size_info, date_info) %>% 
-    mutate(size = size_kb*1024) %>% 
-    mutate(dateUploaded = as.Date(date)) %>% 
-    mutate(formatType = "DATA") %>% 
-    select(-size_kb,-date)
+  mutate(size = as.numeric(size_b)) %>% 
```

Comment:
are we sure the units line up correctly here with what is in the table we eventually join to? based on the variable names it looks like they changed

Reply:
The table we eventually join to is in bytes. In the original slower command for getting sizes, it returned the sizes in kb, then multiplied it by 1024 to convert the kb to bytes. In the new command Nick provided, the sizes are given in bytes, so they don't need to be multiplied by 1024 anymore.